components/provisioner/internal/operations/stages/provisioning/wait_for_agent_to_connect_test.go

```diff
@@ -176,26 +190,53 @@ func TestWaitForAgentToConnect(t *testing.T) {
 		require.Equal(t, 5*time.Second, result.Delay)
 	})
 
-	t.Run("should return error if Compass Connection in Connection Failed state", func(t *testing.T) {
+	t.Run("should return error if Compass Connection in Connection Failed state and runtime reconfigure fails", func(t *testing.T) {
 		// given
 		clientProvider := newMockClientProvider(&v1alpha12.CompassConnection{
 			ObjectMeta: v1.ObjectMeta{Name: defaultCompassConnectionName},
 			Status: v1alpha12.CompassConnectionStatus{
 				State: v1alpha12.ConnectionFailed,
 			},
 		})
+		configurator := &mocks.Configurator{}
+		configurator.On("ConfigureRuntime", cluster, mock.AnythingOfType("string")).Return(apperrors.Internal("test: runtime reconfigure failure"))
 
 		directorClient := &directorMocks.DirectorClient{}
 		directorClient.On("SetRuntimeStatusCondition", cluster.ID, graphql.RuntimeStatusConditionConnected, cluster.Tenant).Return(nil)
 
-		waitForAgentToConnectStep := NewWaitForAgentToConnectStep(clientProvider.NewCompassConnectionClient, nextStageName, 10*time.Minute, directorClient)
+		waitForAgentToConnectStep := NewWaitForAgentToConnectStep(clientProvider.NewCompassConnectionClient, configurator, nextStageName, 10*time.Minute, directorClient)
 
 		// when
 		_, err := waitForAgentToConnectStep.Run(cluster, model.Operation{}, logrus.New())
 
 		// then
 		require.Error(t, err)
 	})
+
```

Comment:
Maybe we can also cover here another test case when the reconfiguration fails 

`configurator.On("ConfigureRuntime", cluster, mock.AnythingOfType("string")).Return(err)`